src/Evolution/Systems/ScalarWave/Equations.cpp

```diff
@@ -24,16 +25,20 @@ void ComputeDuDt<Dim>::apply(
     const gsl::not_null<Scalar<DataVector>*> dt_pi,
     const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*> dt_phi,
     const gsl::not_null<Scalar<DataVector>*> dt_psi,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& d_psi,
     const Scalar<DataVector>& pi,
     const tnsr::i<DataVector, Dim, Frame::Inertial>& d_pi,
-    const tnsr::ij<DataVector, Dim, Frame::Inertial>& d_phi) noexcept {
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& phi,
+    const tnsr::ij<DataVector, Dim, Frame::Inertial>& d_phi,
+    const Scalar<DataVector>& gamma2) noexcept {
   get(*dt_psi) = -get(pi);
   get(*dt_pi) = -get<0, 0>(d_phi);
   for (size_t d = 1; d < Dim; ++d) {
     get(*dt_pi) -= d_phi.get(d, d);
   }
+  const auto constraint = one_index_constraint(d_psi, phi);
   for (size_t d = 0; d < Dim; ++d) {
-    dt_phi->get(d) = -d_pi.get(d);
+    dt_phi->get(d) = -d_pi.get(d) + get(gamma2) * constraint.get(d);
```

Comment:
Just something to think about: I would expect actually doing the math here: `get(gamma2) * (d_psi.get(d) - phi.get(d))` to be more efficient than the allocation in most circumstances.

Reply:
changed